keras_nlp/tokenizers/word_piece_tokenizer.py

```diff
@@ -207,14 +207,25 @@ def __init__(
         self.suffix_indicator = suffix_indicator
         self.oov_token = oov_token
 
-        self._fast_word_piece = tf_text.FastWordpieceTokenizer(
-            vocab=self.vocabulary,
-            token_out_type=self.compute_dtype,
-            suffix_indicator=suffix_indicator,
-            unknown_token=oov_token,
-            no_pretokenization=True,
-            support_detokenization=True,
-        )
+        try:
```

Comment:
One problem about this try-catch is that when RuntimeError is encountered, it will always prompt the user that `oov_token` is problematic (either unset or missing in the vocab). However, it is not only the case that throws RuntimeError, so users may end up seeing the wrong message.

I think here we can explicitly check:
- If `oov_token` is unset. If true, we raise an error.
- If `oov_token` cannot be found in vocab. If true, we prompt users to add the token into their vocab.

Reply:
See my message above. We can just remove the try catch and do our own check, but preserve the error message in this PR which is helpful and just needs copy edits.

We could add a separate check
```
if oov_token is None:
    raise ValueError("oov_token cannot be None")
```
In case users try to manually set oov_token to None I guess. Might come up?

Reply:
This could be the case, if user don't really care about `oov_token` fuss we sought to fix here...

Reply:
@mattdangerw One funny thing here is we posted the review at the exactly same time (during triage meeting), so we got a race condition. 

Yea, we just need to make this check more explicit.